src/tools/tidy/src/deps.rs

```diff
@@ -798,7 +799,7 @@ fn check_runtime_no_duplicate_dependencies(metadata: &Metadata, bad: &mut bool)
             continue;
         }
 
-        if !seen_pkgs.insert(&*pkg.name) {
+        if pkg.name.to_string() != "wasi" && !seen_pkgs.insert(&*pkg.name) {
```

Comment:
I wanted to call out these two changes in particular. Notably the standard library on the wasm32-wasip2 target is picking up a new dependency, `wit-bindgen`. This is developed at https://github.com/bytecodealliance/wit-bindgen/ and basically follows the same development practices of https://github.com/bytecodealliance/wasi-rs, a preexisting dependency of the standard library.

The "wasi" check here is then to allow duplicate dependencies on the "wasi" crate from the standard library since it's intentional that wasm32-wasip1 depends on 0.11 and wasm32-wasip2 depends on 0.14.